plugins/services/src/js/containers/tasks/TasksView.js

```diff
@@ -143,42 +145,63 @@ class TasksView extends mixin(SaveStateMixin) {
   }
 
   getStopButtons() {
-    const {tasks} = this.props;
+    const {service, tasks} = this.props;
     const {checkedItems} = this.state;
 
     if (!Object.keys(checkedItems).length) {
       return null;
     }
+
+    // Only allow restarting the task if the service isn't deploying.
+    const isDeploying = service.getServiceStatus().key
+      === ServiceStatusTypes.DEPLOYING;
     // Only show Stop if a scheduler task isn't selected
     const hasSchedulerTask = tasks
       .some((task) => task.id in checkedItems && task.schedulerTask);
 
     // Using Button's native "disabled" prop prevents onMouseLeave from
     // being correctly called, preventing correct dismissal of the Tooltip.
     //
-    // So we manually disable the button.
+    // So we overwrite the click handlers manually.
+    let handleRestartClick = function () {};
     let handleStopClick = function () {};
 
+    if (!isDeploying) {
+      handleRestartClick = this.handleStopClick.bind(this, 'restart');
```

Comment:
This might also need a rename: ![](https://cl.ly/0c0h3J1g2o3T/Image%202017-01-12%20at%2014.15.35.png)